<launch>
    
    <arg name="ca_camera1_serial" default="16363686"/>
    <!-- arg name="ca_camera1_serial" default="16398533"/ -->
    <arg name="kin_camera1_serial" default="16456195" />
    <arg name="calibrated" default="0" />

    <!-- ========================================================== -->
    <!-- Launch the cameras.                                        -->
    <!-- ========================================================== -->

    <group ns="ca_camera_right">
      <node pkg="nodelet" 
            type="nodelet" 
            name="camera_nodelet_manager"    
            args="manager" />
      <node pkg="nodelet" 
            type="nodelet" 
            name="camera_nodelet"
            args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager" >
            <param name="frame_id" value="camera" />
            <param name="serial" value="$(arg ca_camera1_serial)" />
            <param name="enable_trigger" value="true" />
            <param name="trigger_mode" value="mode14" />
            <param name="video_mode" value="format7_mode5" />
            <param name="auto_exposure" value="false" />
            <param name="auto_shutter" value="false" />
            <param name="auto_gain" value="false" />
            <param name="exposure" value="0" />
            <param name="gain" value="23.5" />
            <!-- shutter speed set to 2 ms faster than trigger rate -->
            <param name="shutter_speed" value="0.012" />
            <!-- Use the camera_calibration package to create this file -->
            <param name="camera_info_url" if="$(arg calibrated)"
              value="file://$(env HOME)/.ros/camera_info/$(arg ca_camera1_serial).yaml" />
      </node>

      <node pkg="nodelet" 
          type="nodelet" 
          name="image_proc_debayer"
          args="load image_proc/debayer camera_nodelet_manager" />
  </group>


    <group ns="kine_camera_1">
      <node pkg="nodelet" 
            type="nodelet" 
            name="camera_nodelet_manager" 
            args="manager" />
      <node pkg="nodelet" 
            type="nodelet" 
            name="camera_nodelet"
            args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager" >
            <param name="frame_id" value="camera" />
            <param name="serial" value="$(arg kin_camera1_serial)" />
            <param name="enable_trigger" value="true" />
            <param name="trigger_mode" value="mode14"/>
            <param name="video_mode" value="format7_mode5" />
            <param name="auto_exposure" value="false" />
            <param name="auto_shutter" value="false" />
            <param name="auto_gain" value="false" />
            <param name="exposure" value="0" />
            <!-- shutter speed set to 2 ms faster than trigger rate -->
            <param name="shutter_speed" value="0.003" />
            <!-- Use the camera_calibration package to create this file -->
            <param name="camera_info_url" if="$(arg calibrated)"
              value="file://$(env HOME)/.ros/camera_info/$(arg kin_camera1_serial).yaml" />
      </node>

      <node pkg="nodelet" 
          type="nodelet" 
          name="image_proc_debayer"
          args="load image_proc/debayer camera_nodelet_manager" />
  </group>

  <!-- ========================================================== -->
  <!-- Launch ledpanels node.                                     -->
  <!-- ========================================================== -->

  <node name="ledpanels" pkg="ledpanels" type="panel_node.py" required="true" />
  <node name="paneldebug" pkg="ledpanels" type="debug_server.py" required="true" />

  <!-- ========================================================== -->
  <!-- Launch the phidgets daq node.                              -->
  <!-- ========================================================== -->

  <param name="/phidgets_daq/update_rate_ms" value="4" type="int"/>
  <param name="/phidgets_daq/ratiometric" value="False" type="bool"/>

  <param name="/phidgets_daq/interpreter_path" value="/home/imager/catkin/src/phidgets_daq/demo/phidgets_daq_interpreter_functions.py" />

  <node name="data_interpreter" pkg="phidgets_daq" type="daq_interpreter.py" output="screen"/>
  <node name="data_acquisition" pkg="phidgets_daq" type="daq.py" output="screen" />
  
  <!-- ========================================================== -->
  <!-- Launch the muscle viewer    .                              -->
  <!-- ========================================================== -->
  <node name="live_viewer" pkg="muscle_imager" type="live_viewer.py"/>
  
  <!-- ========================================================== -->
  <!-- Launch the signal umixer                                   -->
  <!-- ========================================================== -->
  <node name="Unmixer" pkg="muscle_imager"  type="unmixer.py"/>

  <!-- ========================================================== -->
  <!-- Launch the strip chart viewer                              -->
  <!-- ========================================================== -->
  <node name="strip_chart" pkg="muscle_imager" type="strip_chart.py" output = "screen"/>
  
  <!-- ========================================================== -->
  <!-- Launch image blur node.                                    -->
  <!-- ========================================================== -->
  <node name="accumulate_publisher" pkg="muscle_imager" type="accumulate_publisher.py" required="true" />
  
  <!-- ========================================================== -->
  <!-- Launch kinefly                                             -->
  <!-- ========================================================== -->
  
  <!-- include file="$(find Kinefly)/launch/motoman/_main.launch" / -->

</launch>
